process/smartContract/process.go

```diff
@@ -416,6 +416,27 @@ func (sc *scProcessor) resolveBuiltInFunctions(
 		vmOutput = &vmcommon.VMOutput{ReturnCode: vmcommon.UserError, ReturnMessage: err.Error()}
 	}
 
+	if vmInput.CallType != vmcommon.AsynchronousCall && isTooMuchGasProvided(vmInput.GasProvided, vmOutput.GasRemaining) {
```

Comment:
get this if and log trace and output correction in a separate function

Reply:
done